### DISCUSSION TITLE
rabbitmq management plugins support with TLSV1.3

### DISCUSSION THREAD

**Discussion by kaushiksrinivas** [rabbitmq/rabbitmq-server#3757] [Other]
Documentations refer to tlsv1.3 for rabbitmq server only. Is TLSv1.3 officially supported for the management plugin api of the rabbitmq server ?

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
[The same set of settings](https://rabbitmq.com/ssl.html#tls1.3) (namely cipher suites) under `management.ssl` should work (you likely don't want to enable [peer verification](https://rabbitmq.com/ssl.html#peer-verification) for HTTPS clients but that's not specific to TLSv1.3).

**Comment by michaelklishin** [comment]
In a discussion with @lhoguin we have noticed that Cowboy, the HTTP server used by this API, may be injecting [some options by default](https://github.com/ninenines/cowboy/blob/master/src/cowboy.erl#L57) that won't be compatible with TLSv1.3. So when TLSv1.3 is enabled exclusively, Cowboy may need to be extra defensive in future versions.

Otherwise everything should be ready to use TLSv1.3 with any of the protocols supported, as long as there are no option conflicts with earlier versions.

**Comment by michaelklishin** [comment]
[This Erlang 24.1 bug](https://github.com/erlang/otp/issues/5383) is worth mentioning as our team has hit it several times with various protocols. It seems to have been fixed in `24.1.5`.
